packages/dds/tree/src/test/utils.ts

```diff
@@ -423,13 +430,72 @@ export function initializeTestTree(
 	schema: SchemaData = testSchema,
 ): void {
 	tree.storedSchema.update(schema);
-
 	// Apply an edit to the tree which inserts a node with a value
-	tree.runTransaction((forest, editor) => {
-		const writeCursor = singleTextCursor(state);
-		const field = editor.sequenceField(undefined, rootFieldKeySymbol);
-		field.insert(0, writeCursor);
+	const writeCursor = singleTextCursor(state);
+	const field = tree.editor.sequenceField(undefined, rootFieldKeySymbol);
+	field.insert(0, writeCursor);
+}
 
-		return TransactionResult.Apply;
+/**
+ * Runs a transaction on the given forest.
+ * @param forest - the forest to mutate
+ * @param changeFamily - the change family of the edits
+ * @param submitEdit - a callback that provides the change whenever the forest is mutated
+ * @param transaction - the transaction to run. Use the editor passed to this function to mutate the forest.
+ * If it returns an `Abort` result then the transaction will be aborted. Otherwise, it will be committed.
+ * @returns true if the transaction was committed, otherwise false
+ */
+export function runTransactionOnForest<TEditor extends ProgressiveEditBuilder<TChange>, TChange>(
+	forest: IEditableForest,
+	changeFamily: ChangeFamily<TEditor, TChange>,
+	submitEdit: (edit: TChange) => void,
```

Comment:
it seems like either runTransactionOnForest should output an edit, ot it should update the forest (and anchotSet) to the in the modified state. It is confusing to have this submitEdit callback here.

Maybe document that this function leaves the forest in its inital state, and returns a TChange (then make the TChange the return value)?

Reply:
really seems like we should just update our tests to do things the new way, instead of making the old codepath a test helper